### PR TITLE
[CI-v2] Add loadbalancer job with Github Action

### PR DESCRIPTION
**Pull request by EmilienM**
* Deploy Octavia with Devstack
* Run loadbalancer acceptance tests
* Introduce IsReleasesAbove and IsReleasesBelow which return True if a
  release is above the current branch (or below).
* Only test tls_versions in Victoria and beyond, not supported before.
* Only add l7 parameters to QuotaUpdate in Victoria and beyond, not
  supported before.
* Remove all the skips
* Logs failures if any

### CODE REVIEW

File: acceptance/openstack/loadbalancer/v2/quotas_test.go [hunk 1]
```diff
@@ -45,16 +59,21 @@ func TestQuotasUpdate(t *testing.T) {
...
 		Listener:      &originalQuotas.Listener,
 		Member:        &originalQuotas.Member,
 		Pool:          &originalQuotas.Pool,
 		Healthmonitor: &originalQuotas.Healthmonitor,
-		L7Policy:      &originalQuotas.L7Policy,
-		L7Rule:        &originalQuotas.L7Rule,
-	}).Extract()
+	}
+	// L7 parameters are only supported in microversion v2.19 introduced in victoria
+	if clients.IsReleasesAbove(t, "stable/ussuri") {
```
Comment (mandre) [inline acceptance/openstack/loadbalancer/v2/quotas_test.go:70]: Unrelated to your patch but it looks like the implementation of L7 parameters is buggy, they should be pointers according to https://github.com/gophercloud/gophercloud/blob/master/docs/MICROVERSIONS.md#new-response-fields.

Same for tls version above.

File: acceptance/clients/conditions.go [hunk 2]
```diff
@@ -111,3 +111,29 @@ func SkipReleasesAbove(t *testing.T, release string) {
 		t.Skipf("this is not supported above %s, testing in %s", release, current_branch)
 	}
 }
+
+// IsReleasesAbove will return true on releases above a certain
+// one. The result is always true on master release. Releases are named such
+// as 'stable/mitaka', master, etc.
+func IsReleasesAbove(t *testing.T, release string) bool {
```
[the remark this answers is not in the document]
Comment (EmilienM) [reply acceptance/clients/conditions.go:118]: ok, got it. I'll prepare a patch soon.

File: acceptance/openstack/loadbalancer/v2/quotas_test.go (hunk 1 above)
Comment (EmilienM) [reply acceptance/openstack/loadbalancer/v2/quotas_test.go:70]: could you file an issue please?

File: .github/workflows/functional-loadbalancer.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    steps:
+      - name: Checkout Gophercloud
+        uses: actions/checkout@v2
+      - name: Deploy devstack
+        uses: EmilienM/devstack-action@v0.6
+        with:
+          branch: ${{ matrix.openstack_version }}
+          conf_overrides: |
+            enable_plugin octavia https://opendev.org/openstack/octavia ${{ matrix.openstack_version }}
+            enable_plugin neutron https://opendev.org/openstack/neutron ${{ matrix.openstack_version }}
```
[the remark this answers is not in the document]
Comment (mandre) [reply .github/workflows/functional-loadbalancer.yaml:43]: OK, might be related to https://bugs.launchpad.net/neutron/+bug/1955765

**Comment by mandre** [comment]
Let's merge this. We can always refactor IsReleasesAbove() and friends in a follow-up patch, no urgency.

**Pull request merged**